plugins/optimization-detective/optimization.php

```diff
@@ -73,14 +73,18 @@ function od_maybe_add_template_output_buffer_filter() {
  * @return bool Whether response can be optimized.
  */
 function od_can_optimize_response(): bool {
+
+	// Retrieve the sanitized value of the 'REQUEST_METHOD' server variable.
+	$method = filter_input( INPUT_SERVER, 'REQUEST_METHOD', FILTER_SANITIZE_STRING );
+	
 	$able = ! (
 		// Since there is no predictability in whether posts in the loop will have featured images assigned or not. If a
 		// theme template for search results doesn't even show featured images, then this wouldn't be an issue.
 		is_search() ||
 		// Since injection of inline-editing controls interfere with breadcrumbs, while also just not necessary in this context.
 		is_customize_preview() ||
 		// Since the images detected in the response body of a POST request cannot, by definition, be cached.
-		'GET' !== $_SERVER['REQUEST_METHOD'] ||
+		( isset( $method ) && 'GET' !== $method ) ||
```

Comment:
There isn't a need for `isset()` if the `$method` variable was set above, is there? Here it's just checking if it is `null`. There's no string offset going on.

Reply:
I have Updated this

Thanks @westonruter